File: Punto4/fib.c
#include <stdio.h>
#include <time.h>

// Función recursiva de Fibonacci
long long fib(int n) {
    if (n <= 1) return n;
    return fib(n-1) + fib(n-2);
}

int main() {
    int n = 40; // número a calcular
    clock_t start = clock();

    long long resultado = fib(n);

    clock_t end = clock();
    double tiempo = (double)(end - start) / CLOCKS_PER_SEC;

    printf("Fib(%d) = %lld\n", n, resultado);
    printf("Tiempo en C: %f segundos\n", tiempo);
    return 0;
}

File: Punto4/fib.py
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n-1) + fib(n-2)

n = 40  # mismo número
start = time.time()

resultado = fib(n)

end = time.time()
print(f"Fib({n}) = {resultado}")
print(f"Tiempo en Python: {end - start:.6f} segundos")
